segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java

```diff
@@ -39,6 +43,17 @@
      */
     private static final OpStatsLogger GLOBAL_OPERATION_LATENCY = STATS_LOGGER.createStats(MetricsNames.OPERATION_LATENCY);
 
+    /**
+     * Global metric to publish the time elapsed for the recovery of Segment Store containers.
+     */
+    @VisibleForTesting
```

Comment:
I believe the only reason you made this an `AtomicReference` is because of testing. There is another way you can do this:
1. Keep this as a `static final` field.
2. Create a `protected` getter for it; make sure you only use this getter in your code.
3. In your test, create a subclass where you override the getter with whatever you want to supply there.


You need to optimize for production code, not for tests. It's OK if your test has a subclass that exposes some fields - it's much better than exposing them unnecessarily or adding unneeded complexity for all cases.